userspace/libsinsp/filter_compare.cpp

```diff
@@ -583,9 +583,36 @@ bool flt_compare_ipv6net(cmpop op, const ipv6addr* operand1, const ipv6net* oper
 // flt_cast takes a pointer to memory, dereferences it as fromT type and casts it
 // to a compatible toT type
 template<class fromT, class toT>
-static inline toT flt_cast(const void* ptr) {
-	fromT val;
-	memcpy(&val, ptr, sizeof(fromT));
+static inline toT flt_cast(const void* ptr, uint32_t len) {
+	fromT val{};
+	/*
+	 * In big endian systems, we need to
+	 * make sure that we copy the right bytes
+	 * when len > sizeof(fromT).
+	 * This is an edge case that should only happen
+	 * for `evt.rawarg.*` fields.
+	 */
+	uint8_t shift = 0;
+#ifdef __s390x__
+	// NOTE: c++20 has `constexpr (std::endian::native == std::endian::big)`
```

Comment:
It would be huge if we were able to bump our c++ standard to 20 and make use of this; we would be able to check for endianness at compile time.